cudaextender/src/ungapped_xdrop_kernels.cuh

```diff
@@ -60,21 +60,50 @@ __global__ void compress_output(const int32_t* d_done,
                                 int num_hits);
 
 // Binary predicate for sorting the ScoredSegmentPairs
+struct scored_segment_pair_equal
+{
+    __host__ __device__ bool operator()(ScoredSegmentPair x, ScoredSegmentPair y)
+    {
+            return ( ( 
+                        (x.seed_pair.target_position_in_read - x.seed_pair.query_position_in_read) == (y.seed_pair.target_position_in_read - y.seed_pair.query_position_in_read) 
+                        ) 
+
+                    &&  
+
+                    ( 
+                     ( 
+                      (x.seed_pair.target_position_in_read >= y.seed_pair.target_position_in_read) 
+                      && 
+                      ( (x.seed_pair.target_position_in_read + x.length) <= (y.seed_pair.target_position_in_read + y.length) )  
```

Comment:
This looks a bit dangerous to me. If I understand it correctly, you add the full length of the extension to the target position, instead of just the right or left extension length.
This might be an artificial case, but assume I have
```
x.seed_pair.target_position_in_read = 30;
x.length = 5; (because it extends from 30-34)
y.seed_pair.target_position_in_read = 20;
y.length = 20; (because it extends from 1-20)
```
=> `(x.seed_pair.target_position_in_read >= y.seed_pair.target_position_in_read) = true` and `(x.seed_pair.target_position_in_read + x.length) <= (y.seed_pair.target_position_in_read + y.length) = true`
even though these are disjoint extensions.
Does this not matter or can it never happen?

Reply:
I don't think I understand correctly. But just for context -- Going from the seed pair to segment pair, `segment_pair.target_position_in_read = seed_pair.target_position_in_read - left_extension_length`. Also, for this equality comparison, we look at the segments pairs on the same diagonal i.e. `(x.seed_pair.target_position_in_read - x.seed_pair.query_position_in_read) == (y.seed_pair.target_position_in_read - y.seed_pair.query_position_in_read)`.

Reply:
Hm. I'm still a bit confused about this one.
Could you explain me in words: when would you consider two ScoredSegmentPairs to be equal?
We can also hop on a short call if that makes it easier to explain.

Reply:
We discussed offline and I understand what my misconception was now. Makes sense now. Thanks!